packages/medusa/src/api/utils/middlewares/products/normalize-data-for-context.ts

```
@@ -0,0 +1,74 @@
+import { AuthenticatedMedusaRequest } from "../../../../types/routing"
+import { refetchEntities, refetchEntity } from "../../refetch-entity"
+
+export function normalizeDataForContext() {
+  return async (req: AuthenticatedMedusaRequest, _, next: NextFunction) => {
+    // If the product pricing is not requested, we don't need region information
+    let withCalculatedPrice = req.remoteQueryConfig.fields.some((field) =>
+      field.startsWith("variants.calculated_price")
+    )
+
+    // If the region is passed, we calculate the prices without requesting them.
+    // TODO: This seems a bit messy, reconsider if we want to keep this logic.
+    if (!withCalculatedPrice && req.filterableFields.region_id) {
+      req.remoteQueryConfig.fields.push("variants.calculated_price.*")
+      withCalculatedPrice = true
+    }
+
+    if (!withCalculatedPrice) {
+      return next()
+    }
+
+    // Region ID is required to calculate prices correctly.
+    // Country code, and optionally province, are needed to calculate taxes.
+    let regionId = req.filterableFields.region_id
+    let countryCode = req.filterableFields.country_code
+    let province = req.filterableFields.province
+
+    // If the cart is passed, get the information from it
+    if (req.filterableFields.cart_id) {
+      const cart = await refetchEntity(
+        "cart",
+        req.filterableFields.cart_id,
+        req.scope,
+        ["region_id", "shipping_address.*"]
+      )
+
+      if (cart?.region_id) {
+        regionId = cart.region_id
+      }
+
+      if (cart?.shipping_address) {
+        countryCode = cart.shipping_address.country_code
+        province = cart.shipping_address.province
+      }
+    }
+
+    // Finally, try to get it from the store defaults if not available
+    if (!regionId) {
+      const stores = await refetchEntities("store", {}, req.scope, [
+        "default_region_id",
+      ])
+      regionId = stores[0]?.default_region_id
+    }
```

sugg: If region_id is not found from the cart or the query param, we should either throw or return without prices. If both the cart or query fail, we're already in a weird place. Picking from the store might put the query/customer in a weirder place if its a multi region store 😄 

I prefer the throw because I'd rather know an error is popping up as a dev so that I can fix it than silently returning.

---

If we do that, then the store shouldn't have a default region at all, do we agree on that? Let's see what others think about this.

---

> If we do that, then the store shouldn't have a default region at all

Couldn't there be other use cases for a default region on the store? I am not it was added for the sole purpose of having a fallback for price calculations